Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -2187,6 +2187,79 @@ extension EventLoopFuture {
     }
 }
 
+// MARK: assertion
+
+extension EventLoopFuture {
+    /// Returns a new `EventLoopFuture` that asserts the original future's success.
+    ///
+    /// If the original future fails, it triggers an assertion failure, causing a runtime error during development.
+    @inlinable
+    public func assertSuccess() -> EventLoopFuture<Value> {
```

Comment:
here and in the other functions we should take `file: ... =  #file, line = ... #line` and pass that to the (assertion/precondition)Failure functions

Reply:
Done to all!